packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      checkOperationResult(updateResult, { ...todoUpdated, customGroups: null, privateContent: null }, updateResultSetName);
+
+      // non-admin can read the private protected field
+      const getResult = await todoHelperNonAdmin.get(
+        {
+          id: todoWithAllowedFields['id'],
+        },
+        allowedResultSet,
+        false,
+      );
+      checkOperationResult(getResult, todoUpdated, `get${modelName}`);
+
+      const listTodosResult = await todoHelperNonAdmin.list({}, allowedResultSet, `list${modelName}`, false);
+      checkListItemExistence(listTodosResult, `list${modelName}`, todoWithAllowedFields['id'], true);
+
+      // unless all fields have same Auth rules as the model, delete is expected to fail
+      await expect(
+        async () => await todoHelperNonAdmin.delete(`delete${modelName}`, { id: todoWithAllowedFields['id'] }, allowedResultSet),
+      ).rejects.toThrowErrorMatchingInlineSnapshot(`"GraphQL error: Not Authorized to access delete${modelName} on type Mutation"`);
+
+      const todoRandom = {
+        ...todoWithAllowedFields,
+        id: Date.now().toString(),
+      };
+      const todoRandomUpdated = {
+        ...todoWithAllowedFields,
+        id: todoRandom.id,
+      };
+      const actorClient = getConfiguredAppsyncClientCognitoAuth(graphQlEndpoint, region, userMap[userName2]);
+      const subTodoHelper = createModelOperationHelpers(actorClient, schema)[modelName];
+
+      const onCreateSubscriptionResult = await subTodoHelper.subscribe(
+        'onCreate',
+        [
+          async () => {
+            await todoHelperAdmin.create(createResultSetName, todoRandom, allowedResultSet);
+          },
+        ],
+        {},
+        allowedResultSet,
+        false,
+      );
+      expect(onCreateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onCreateSubscriptionResult[0],
+        { ...todoRandom, customGroups: null, privateContent: null },
+        `onCreate${modelName}`,
+      );
+
+      const onUpdateSubscriptionResult = await subTodoHelper.subscribe(
+        'onUpdate',
+        [
+          async () => {
+            await todoHelperAdmin.update(`update${modelName}`, todoRandomUpdated, allowedResultSet);
+          },
+        ],
+        {},
+        allowedResultSet,
+        false,
+      );
+      expect(onUpdateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onCreateSubscriptionResult[0],
+        { ...todoRandomUpdated, customGroups: null, privateContent: null },
+        `onUpdate${modelName}`,
+      );
+
+      const onDeleteSubscriptionResult = await subTodoHelper.subscribe('onDelete', [], {}, allowedResultSet, false);
+      expect(onDeleteSubscriptionResult).toHaveLength(0);
+    });
+
+    test.skip('user part of group in allowed custom groups list of a record cannot perform CRUD operations on public protected field', async () => {
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket